DTAdmin/Models/DataManager.swift

```diff
@@ -166,6 +166,85 @@ class DataManager: HTTPManager {
         }
     }
     
+    func getListOfQuestions(forEntity typeEntity: Entities, entityId: String, quantity:UInt, fromNo index: UInt, completionHandler: @escaping (_ listEntity: [Any]?, _ error: String?) -> ()) {
+        let indexString = String(index)
+        let quantityString = String(quantity)
+        guard let request = self.getURLReqest(entityStructure: typeEntity, type: TypeReqest.GetRecordsRangeByTest, id: entityId, limit: quantityString, offset: indexString) else {
+            let error = "Cannot prepare header for URLRequest"
+            completionHandler(nil, error)
+            return
+        }
+        self.getResponse(request: request) { (list, error) in
+            if let error = error {
+                StoreHelper.logout()
+                DispatchQueue.main.async {
+                    completionHandler(nil, error)
+                }
+            }
+            guard let  json = list as? [[String: Any]] else {
+                let error = NSLocalizedString("Response is empty", comment: "No data in server response")
```

Comment:
Bad error message. Consider a case when the server returns one element instead of an array.

Reply:
He always return an array, in case with one element it will be array with one element